### ISSUE TITLE
Prime blob or prime line can exceed maximum extrusion (esp. with small nozzles)

### ISSUE THREAD

**Issue by david-antos** [Rat-Rig/V-CoreOS#113]
### What happened

Switching to 0.25mm nozzle, I got
```
Move exceeds maximum extrusion (0.289mm^2 vs 0.250mm^2)
See the 'max_extrude_cross_section' config option for details
```
during prime blob. I tried prime line as well, which pushes even more filament, resulting in the same error.
I have correct nozzle diameter in printer.cfg (as well as in my slicer setting, which is probably not relevant as priming is controlled solely by printer configuration). In addition to that, when priming is set to False, there are no issues with the nozzle (suggesting my setup is likely fine otherwise).

What is even worse, the error is displayed, printing stops, and both bed and extruder heaters stay on.

### What did you expect to happen

Should the maximum extrusion be exceeded, RatOS should at least end up in shutdown (not keeping the extruder in temperature).

Priming the nozzle should be possible regardless of nozzle size.

### How to reproduce

Start the print under conditions described in the "what happened" section. The process is deterministic.

### Additional information
None

_Edited: my initial reading of the code was incorrect, so the proposed solution is not relevant._

**Comment by miklschmidt** [comment]
Prime blob needs extrusion calculation based on nozzle size for sure. It should however still work fine with 0.25mm nozzles if you leave the nozzle_diameter at 0.4, i'd use that as a workaround for now.

The only thing nozzle_diameter is used for in klipper is to calculate the threshold to throw that exact error.

> What is even worse, the error is displayed, printing stops, and both bed and extruder heaters stay on.

That's how klipper behaves on non fatal errors. You'd have to raise an issue over there.

**Comment by miklschmidt** [comment]
in 2.1 (still unreleased), primeblob extrusion is now scaled by nozzle diameter (which is set in the configurator).

**Comment by miklschmidt** [comment]
Closed following 2.1.0-RC1
